### ISSUE TITLE
Configure an album via a simple text file

### ISSUE THREAD

**Issue by oparoz** [owncloud/gallery#93]
Maybe you want to include a an album description? #85 
Or you want to make your sorting preferences for this album permanent? #91 
You're required to add a copyright notice to your public gallery?
etc.

A simple [yaml](https://en.wikipedia.org/wiki/YAML) file could be looked for and the information used to format the album being currently looked at.

The advantage of using a file, vs using the database is that you'll get the exact same view on another ownCloud using the same app. You can also quickly change the settings on your desktop, without having to log into ownCloud itself.

**The documentation is available here:** https://github.com/interfasys/galleryplus/blob/master/documentation/Album%20configuration.md
### Feasibility study
- [x] Collect requirements
- [x] Calculate effort required - 2 days
- [x] Build proof of concept
### Development
- [x]  Secure funding - Sponsored by [Cute cut](http://www.cute-cut.ch)
- [x]  Finalise specifications
### QA
- [x] Define tests
- [x] Pass all tests
### Documentation
- [x] Create documentation
- [x] Deliver training

**Comment by oparoz** [comment]
I'm thinking of calling the config file `gallery.cnf`
`conf`, but as a 3 letter extension.

I did not want to use a hidden file because it needs to be editable from within oC. (It's currently not, but I've submitted a PR for it.)

I'm not going to bother with the nomedia entry atm, simply because 99% of people will not need to name it differently and because it's not portable.

**Comment by deMattin** [comment]
I could not figure out, how to use it.
Is anything more needed to do?
I switched to album-info BR, created file "gallery.cnf" and modded it.
But I see no difference.
Do you only need one file in a root folder to get all sub folders or do you have to set this file in any folder?
Which functions or settings do already work?
Which parameters are allowed?

Your changes in code are so much to read, that it is nearly impossible to figure it out.

Btw: customizing ".nomedia" should remain but set to default ".nomedia" if nothing other is set on album-info.

**Edit:**
Uhh - now sorting of pictures and info works.
Sorting of folder doesn't seem to work, yet ...

**Edit2:**
Firs little issue: No "german umlauts" like äüö allowed in description ...

**Comment by oparoz** [comment]
It's one config file per folder.

The text file has to look like the one above, except the nomedia line.
Everything should work.

You need to make sure you've cleared the asset cache, as usual, so that the new code can be loaded. Maybe restarting PHP if using PHP-FPM.

In the debug, window, network tab, the call to `files` should contain all the image objects as well as the album permissions

**Comment by deMattin** [comment]
It needed some time to recognize the files - I have to do more than one reload.
I deactivate assets for testing completely since last discussion in core git about this! ;)

As I edited above: No "German Umlauts" like äüö allowed in description ...

File sorting works. I tried name reverse but had to set it reverse:

> sorting: name
> sort_order: asc

Seems you switched ascending (A-Z) and descending (Z-A) in code.

But folder sorting with same gallery.cnf seems not to work at all.

**Edit:**
Is it planned to inherit configs so it would only be neccessary to set one file in root to set this for all subfolders, too?

**Edit2:**
Error with copyright info if you link a simple text-file:

```
Die Webseite unter
https:/[domain]/owncloud/index.php?redirect_url=%2Fowncloud%2Findex.php%2Fapps%2Fgalleryplus%2Fdownload%3Ffile%3DFotos%2Fgallery.copyright.md
hat zu viele Umleitungen erzeugt.
Das Löschen Ihrer Cookies für diese Website oder das Zulassen von Drittanbieter-Cookies
könnte das Problem möglicherweise beheben. Sollte dies nicht der Fall sein, liegt eventuell
ein Serverkonfigurationsfehler und kein Problem mit Ihrem Computer vor.
```

**Comment by oparoz** [comment]
Regarding sorting, you're right, I thought the default was called descending, even if I thought it was counter-intuitive. 

I don't have any issues with special characters, even on IE ;). Maybe it's a PHP problem? Or maybe you've used HTML characters?
If you could provide a sample readme.md perhaps?

I thought about organising sorting like the nomedia, but decided against it as it would make it too complex to keep track of things. You need sorting in every folder whereas you only need to hide some folders.

**Comment by oparoz** [comment]
And album sorting is untouched because I don't have the album mtime.

**Comment by deMattin** [comment]
It has been a "classic" error with the umlauts:
I created the file with windows editor -> so it has been ANSI and not UTF8 ...
Changed it to UTF8 and it works now!

Album mtime?
I like to sort them on **name**!
It would be nice to get this working, too.

Here's my gallery.cnf (with the reverse-reverse) sorting:

```
---
description: gallery.info.md
copyright: Copyright 2003-2015 MK, Germany
copyright_link: gallery.copyright.md
sorting: name
sort_order: asc
```

One error left:
click on copyright-link causes above quoted error in chrome.

**EDIT**:

> I thought about organising sorting like the nomedia, but decided against it as it would make it too complex to keep track of things. You need sorting in every folder whereas you only need to hide some folders.

Hmm - would it then be possible to define **one** gallery.cnf in the account root to set defaults?
It will be very hard to copy the cnf in any folder if you want to have another default setting!
This way gallery+ only has to check root folder and the current folder with cnf in current folder overwriting "default" cnf.

**Comment by oparoz** [comment]
I wonder if there is something we can do about ANSI files... Let's see if more people have issues.

mtime as in modification time. We just return images and the script builds albums based on the paths, so it has not information about the albums at all.
It would probably be possible to collect more information ,but that requires more than just adding a variable.

Name sorting should work, but items are sorted differently than on the Files side. I get the wrong results on the Files app.

It seems the copyright link doesn't work on public galleries, but are you seeing something else?

**Comment by deMattin** [comment]
I only see the error message. Nothing is displayed.  But it's from the account and not a public gallery. 
Let's go further tomorrow. Have to go to work in 6 hours. ;-)

**Comment by oparoz** [comment]
You didn't add the error here, so I don't know what it says.

**Comment by oparoz** [comment]
If the file is not found, there is a message about an internal server error

**Comment by deMattin** [comment]
The error is some postings up as "Edit":

Chrome error with copyright info if you link a simple text-file:
Too many redirections ...

```
Die Webseite unter
https:/[domain]/owncloud/index.php?redirect_url=%2Fowncloud%2Findex.php%2Fapps%2Fgalleryplus%2Fdownload%3Ffile%3DFotos%2Fgallery.copyright.md
hat zu viele Umleitungen erzeugt.
Das Löschen Ihrer Cookies für diese Website oder das Zulassen von Drittanbieter-Cookies
könnte das Problem möglicherweise beheben. Sollte dies nicht der Fall sein, liegt eventuell
ein Serverkonfigurationsfehler und kein Problem mit Ihrem Computer vor.
```

**Comment by deMattin** [comment]
Another weird thing.
I checked right now at work and here I am behind a **proxy**.
The folder with the gallery.cnf here shows the "no pictures found" message of gallery +.

If I deactivate "gallery.cnf" by renaming it, pictures are shown.

**Comment by oparoz** [comment]
It's funny how Github caching works... I had the same problem in another thread.
Anyway, that error should be gone now.

Regarding the proxy, try a ctrl+f5, just to see if it's not a caching issue.
If not, look at the browser debugger. You're probably not getting the files list and I'd like to know why. There is probably a 4xx or 5xx error.

And I think I will need to implement hierarchy because the defaults don't work for some people, so they need to be able to configure their whole configuration with one file.

**Comment by deMattin** [comment]
Proxy (starting gallery+ in a subfolder with gallery.cnf defined):
Following error is shown in Chrome browser console:

> GET https://[domain]/owncloud/index.php/apps/galleryplus/files?location=Bilder 500 (Internal Server Error)  jquery.min.js?v=4e066f04f1a7ef7e5d3bc5f2629d463d:6
> 
> x.ajaxTransport.send   jquery.min.js?v=4e066f04f1a7ef7e5d3bc5f2629d463d:6
> x.extend.ajax   jquery.min.js?v=4e066f04f1a7ef7e5d3bc5f2629d463d:6
> x.each.x.(anonymous function)   jquery.min.js?v=4e066f04f1a7ef7e5d3bc5f2629d463d:6
> x.extend.getJSON   gallery.js?v=4e066f04f1a7ef7e5d3bc5f2629d463d:50
> Gallery.fillAlbums   gallery.js?v=4e066f04f1a7ef7e5d3bc5f2629d463d:543
> (anonymous function)   jquery.min.js?v=4e066f04f1a7ef7e5d3bc5f2629d463d:4
> x.Callbacks.cj   query.min.js?v=4e066f04f1a7ef7e5d3bc5f2629d463d:4
> x.Callbacks.p.fireWith   jquery.min.js?v=4e066f04f1a7ef7e5d3bc5f2629d463d:4
> x.extend.ready   jquery.min.js?v=4e066f04f1a7ef7e5d3bc5f2629d463d:4 q

The "Info", "Share" and "Switch to file app" Buttons come up very short and then they go away and the message is shown.
Do you have another idea to check than proxy relation - which I'm not able to control?

I made a git pull right now and got changes on servicecontroller.php and gallery.js

The redirect bug I can only test at home because of this proxy thing ... ;)

**Comment by oparoz** [comment]
I suspect it's not a proxy issue. Maybe you could try from a mobile?

**Comment by deMattin** [comment]
At home now and you are right: It has nothing to do with proxy because it doesn't work here, too.

What's that?! Yesterday it worked! :disappointed:

**Comment by oparoz** [comment]
Maybe a permission issue, but if that's the case I need to ignore the file instead of breaking the whole thing :D

**Comment by oparoz** [comment]
But check the logs as it will tell you what went wrong.

**Comment by deMattin** [comment]
This is in any case a good idea.

But I found the failure: UTF8 with BOM causes this.
Changed file format to UTF8 without BOM and it works.

I think, I changed all formats yesterday because of umlauts issue in copyright and info files and didn't test it finally.

In both other files it doesn't matter if file format is UTF8 with or without BOM, but in gallery.cnf it seems to be essential.

**Comment by deMattin** [comment]
An click on link of Copyright now downloads the linked Copyright info file.

**Comment by oparoz** [comment]
It would be great if you could find what the exact error was. I'm wondering if it's an oC or a parser problem.

**Comment by deMattin** [comment]
Seems to be a parser error (ownCloud log):

> Exception: {"Message":"Unable to parse at line 1 (near \"\ufeff---\").","Code":0,"Trace":"#0 \/var\/www\/owncloud\/apps\/galleryplus\/vendor\/symfony\/yaml\/Symfony\/Component\/Yaml\/Yaml.php(67): Symfony\Component\Yaml\Parser->parse('\xEF\xBB\xBF---\r\ndescrip...', false, false)\n#1 \/var\/www\/owncloud\/apps\/galleryplus\/controller\/servicecontroller.php(289): Symfony\Component\Yaml\Yaml::parse('\xEF\xBB\xBF---\r\ndescrip...')\n#2 \/var\/www\/owncloud\/apps\/galleryplus\/controller\/servicecontroller.php(271): OCA\GalleryPlus\Controller\ServiceController->hasAlbumConfig(Object(OC\Files\Node\Folder))\n#3 \/var\/www\/owncloud\/apps\/galleryplus\/controller\/servicecontroller.php(148): OCA\GalleryPlus\Controller\ServiceController->getAlbumInfo(Object(OC\Files\Node\Folder), '\/mk\/files\/')\n#4 [internal function]: OCA\GalleryPlus\Controller\ServiceController->getFiles('Bilder')\n#5 \/var\/www\/owncloud\/lib\/private\/appframework\/http\/dispatcher.php(158): call_user_func_array(Array, Array)\n#6 \/var\/www\/owncloud\/lib\/private\/appframework\/http\/dispatcher.php(86): OC\AppFramework\Http\Dispatcher->executeController(Object(OCA\GalleryPlus\Controller\ServiceController), 'getFiles')\n#7 \/var\/www\/owncloud\/lib\/private\/appframework\/app.php(97): OC\AppFramework\Http\Dispatcher->dispatch(Object(OCA\GalleryPlus\Controller\ServiceController), 'getFiles')\n#8 \/var\/www\/owncloud\/lib\/private\/appframework\/routing\/routeactionhandler.php(44): OC\AppFramework\App::main('ServiceControll...', 'getFiles', Object(OC\AppFramework\DependencyInjection\DIContainer), Array)\n#9 [internal function]: OC\AppFramework\routing\RouteActionHandler->__invoke(Array)\n#10 \/var\/www\/owncloud\/lib\/private\/route\/router.php(250): call_user_func(Object(OC\AppFramework\routing\RouteActionHandler), Array)\n#11 \/var\/www\/owncloud\/lib\/base.php(767): OC\Route\Router->match('\/apps\/gallerypl...')\n#12 \/var\/www\/owncloud\/index.php(36): OC::handleRequest()\n#13 {main}","File":"\/var\/www\/owncloud\/apps\/galleryplus\/vendor\/symfony\/yaml\/Symfony\/Component\/Yaml\/Parser.php","Line":285}

I only changed file format of gallery.cnf from UTF8 with BOM to UTF8 w/o BOM to get this error message.

**Comment by oparoz** [comment]
Thanks. I might add a BOM remover then, even if the instructions indicate what the specs of the file should be.

**Comment by deMattin** [comment]
Instructions?!? :wink:

**Comment by deMattin** [comment]
Is seems to be intent to download and not to show the copyright info (tested with .md and .txt).
Would it be possible to recognize, if you set a URL (like http://blabla.de/copyinfo.htm) here to just show the copyright info and not to download it?
It doesn't neccessarily need to be a html file in owncloud account and can be a public html-page.

**Comment by deMattin** [comment]
I played a little with markdown and I think "Copyright" file is not neccessary if you want to give a link to a page, like I asked above.
It's as simple as defining a link and some text with markdown in readme.md (description file) and not using the described copyright-file and tags in gallery.cnf.
This way you have both possibilitys: Force download of a Copyright-file from picture folder or define anything in description file using markdown.
I like it! :+1: 

Only thing now missing is the "default" configuration ...

Btw: I'm missing a simple linebreak in markdown.

**Comment by deMattin** [comment]
Ok, another thing with markdown:
Is there a possiblity to set Spacing between headlines and following text?
The space here is as big as the space between normal text and the next following headline.
This looks a bit weird and isn't like that in http://www.markitdown.net/markdown where header are more narrow to the following and corresponding text.
![image](https://cloud.githubusercontent.com/assets/5565503/6788247/85777c7c-d198-11e4-9bf7-aab04ecfd844.png)

**Comment by deMattin** [comment]
As I'm on theming of my oC, I found, that, here are differences between gallery+ master and this branch in older and unrelated files. I hope, there are not more differences ...:
I don't understand, how this can occur in git!

Wrong alignment in gallerybutton.css in album-info branch:

**master** (as it should be):

``` javascript
#openAsFileListButton {
    float: left;
    margin-top: 5px;
...
```

This **album-info** branch:

``` javascript
#openAsFileListButton {
    float: left;
    margin-top: 4px;
...
```

**Comment by oparoz** [comment]
Regarding the copyright, don't forget that people distribute archives of folders. They want all the files to be available in the same folder. So, for them, it's easier to just add the name of the file to make it automatically available.
I will accept markdown as the copyright text. That way people can add a link if they simply use licenses such as CC.

Regarding the spacing, look at he OP. There is a lot of spacing below each title.
I can certainly use a different stylesheet, but it has to be easily installable via bower.

And regarding the CSS, I had to update it, so that elements are properly aligned, but maybe I should do it on the parent class.
Master will be upgraded to have a 4px margin, unless someone can show me a way to fix things differently.

**Comment by deMattin** [comment]
Copyright: All ok, as it is now and with markdown even better!

Spacing: I don't understand, where I should have a look at. It would be nice to generally halfen the space between headlines and following text. But if it is a problem, I can live with it how it is.

CSS: Master does it the right way now. 5px is the recommendet setting!
It's the wrong setting in album-info branch.
Won't a commit in branch to 5px and merge in master fix this difference?

**Comment by oparoz** [comment]
Spacing: Check https://github.com/interfasys/galleryplus/blob/album-info/css/github-markdown.css . You can even replace the whole stylesheet. The spacing doesn't bother me, but we'll see once this is released.

CSS: Master and core do it the wrong way. Buttons are not vertically centred, the top margin should be 4px... but it's true that the Gallery button now shows that the other ones are wrong, so maybe it's best to wait for core to fix the problem...
Put a piece of paper on your screen, under the text written in the breadcrumb. You'll see that "New" is too low.

**Comment by deMattin** [comment]
The core css is very hard to maintain, I think.
With my theming of file view (less waste of vertical space) I recognized, that there are much unneeded and hardcoded values, that depend on each other and so you have to change things on many places to only alter one visual setting.
But this is another story and I'm no css expert - maybe it all has its reason, that I only don't see ... ;)

Where does this css _github-markdown.css_ come from?
It doesn't look very optimized or reviewed.
The css styles for `.markdown-body h..` are definied in line 117 and again in line 234 overwriting the previous one.
Seems, that there are at least 2 css-style files simply merged together.

The following modification defines spacing between headline and following text as big as linespacing in headline - which always looks good in my opinion.

If you set in line 241 (second definition):

> margin-top: 1em;
> margin-bottom: 0.0em;

instead of

> margin-top: 1em;
> margin-bottom: 16px;

It looks like this - with headlines better grouped with following text:
![image](https://cloud.githubusercontent.com/assets/5565503/6792132/574cb520-d1b5-11e4-82ae-72c7a479489f.png)

Also paragraph spacing of normal text can be optimized - it's also a bit too large in relation to the line height (you can see it in chapter "Bilder 3").
But I don't want to have a css and "style" discussion again - I recently had one in core git and obviously big spacing with less content is the thing anybody but me wants to have ...

As it is all themeable with oC theming, it doesn't bother me, if you leave css as it is (also with the double definitions) - or let it review by a member, knowing (and liking) the "oC styleguide". :wink:

**Comment by oparoz** [comment]
CSS is a nightmare to maintain, especially when you have optional components such as apps.
It would be much better with SCSS, but it's considered a barrier to entry.

The markdown CSS comes from here: https://github.com/sindresorhus/github-markdown-css
Let them know if you find bugs :)
And as I mentioned earlier, I'm not set on this stylesheeet. Maybe it's too Githuby and something else is needed for a gallery app. I just prefer something available via bower so that I don't have to maintain it.

I've set it back to 5px since it's the standard, but as you can see on here, it's too close to the bottom.
![image](https://oc8demo.interfacloud.com/index.php/s/GhqEzzckGPv4Alh/download)

**Comment by deMattin** [comment]
I will have a closer look at the github-markdown.css and open an issue there.
Because I think it's already used in some projects, changes surely have to be made careful.
So I agree, not to patch this and as this css is really easy to theme in oC if you want to, it's not a problem. 

The alignment error is one of the things, I talked about:
Hard coded values, borders and paddings often are used in oC css, where it would be better to use alignment.
I think the css designer used a ready to use tool for creating the base css. This way you often get such absolute definitions for spacings to align objects instad of better to be used relative or alignment definitions.
And from release to release the code get's bigger and it will become harder to maintain it, I think.
I have no idea, how to solve this easily, because you have to rewrite, check and optimize all css code and this will be hard to test because I think, there are also some tricky definitions in code to solve special problems.

SCSS and SASS look very nice but also need a complete rework to integrate really well.
You should or have to start from scratch and this only can be done by core maintainers who know more about internal roadmaps and discussions in team.

Btw: Nice buttons for sorting ... :wink:

**Comment by deMattin** [comment]
I have an idea and want to ask, what you think about it.
By now we have sorting options that are assigned to files and folders at same time with different issues:
- folder can't be sorted, if you sort by date
- folder and file follow the same sorting option
- ...

What do you think about defining seperate tags in gallery.cnf for folder sorting and file sorting?
Something like this:

> sorting-file: date
> sort_order-file: des
> sorting-folder: name
> sort_order-folder: asc

You use separate variables in code, I think, and it won't be many code changes but some more possibilities for individual sorting settings.

**Usage example:**
I have picture folder named chronological (folder name sorting reverse prefered) and folder named thematical (name sorting prefered)  but in both I would like to have reverse sorting of filename or date.
With above described enhancement, it would be possible to realise this.

**Comment by oparoz** [comment]
The problem with your example is that it's going to be impossible to sort using simple buttons, but maybe this can be done if a sorting panel is ever introduced where we can tick various options, etc.

Once this is merged into master, open a ticket as a feature request

**Comment by deMattin** [comment]
I see no issue with the button resorting. In this case it simply overrides temporary the values for both settings - folder and file sorting.
I see no mandatory reason to give any oportunity in config also in a corresponding panel setting.

The buttons should not set the values in config file permanent - independent of my proposial!

Or maybe it could be realized as a "hidden and not (yet) dokumented feature", to define this in gallery.cnf? :smile:

**Comment by oparoz** [comment]
Don't forget that the gallery also has a public side. It has to work for people who have never used the app before. They will get confused and think it's broken. It's also out of the scope of what I was asked to do by the client.

The best thing to do in your case is to fork it once this is merged and to customise it. You would only need to modify the Javascript to take those new settings into consideration.
But do open a ticket. The more people talk about a feature, the easier it is to understand and define.

**Comment by deMattin** [comment]
Forking will be no problem (sadly github has no rebase feature or I only didn't found it, yet) - and customizing js is also possible with oC theming.
I will wait and decide later, what will be best to do.

You are right - for public galleries this will confuse, if you set a not in GUI setable default sorting order.

For public galleries I can imagine a separate default (could be as simple as ignoring special folder sort) or not to interprete special and different sorting at all and always sort by name on default switchable by buttons in browser session.

But not making it to complicated is also a relevant point and so I wait and let you finalize actual features (and test them).

Btw.: I also like the option in linux nautilus to sort folder interwoven with files alphabetically - but that's even for me to much here and could be rather a nice option in file view ... ;)

**Comment by oparoz** [comment]
I'm sure people would scream "Where are my albums?!?" if the Nautilus feature was implemented ;), but anything is possible if we have a nice sorting panel.

**Comment by oparoz** [comment]
The feature is complete now, but I'm not a 100% happy with the sliding panel, so I need to see if I can improve it...

**Comment by deMattin** [comment]
Should this feature already fully work?

> **Sorting**
> After careful consideration, it has been decided that :
> - the script would look for sorting configurations in parent folders in order to make it easier to define a global sorting order. Exception configurations would have to be defined in sub-folders. More testing is required in order to determine if that's a solution which would cover most use cases

Actual branch `master` or `sorting-buttons` both seem to inherit the sorting but not the info.
Is it planned to inherit this, too - would be nive to have ...

**Comment by oparoz** [comment]
It works for sorting yes and the buttons auto-update to show you what's in use.
I don't think it will work for descriptions. You describe your gallery as "This is a collection of albums filled with pictures taken in various locations" and then reach a leaf album with just pictures of a party on the beach. The description would not work and you would realise that you would need to add descriptions to all folders in order to avoid seeing that message.

Maybe I could introduce a keyword like "inherit", so that the same description can be used in sub-folders.

**Comment by oparoz** [comment]
It's best to discuss new additions in separate tickets. Inheritance is discussed here: https://github.com/interfasys/galleryplus/issues/107

**Comment by deMattin** [comment]
Didn't see it separate - will go further there ... thanks

**Comment by oparoz** [comment]
That's because I've just created the ticket ;)

**Comment by oparoz** [comment]
**Warning - The configuration has changed**
https://github.com/interfasys/galleryplus/blob/master/documentation/Album%20configuration.md

**Comment by oparoz** [comment]
**Warning - The configuration has changed again**

**Comment by deMattin** [comment]
Seems to be something wrong in your manual, how to use the description file:

> information:
>   description_link: This is an album description
>   description_link: readme.md

I think, it has to be:

> information:
>   description: This is an album description
>   description_link: readme.md

And it should be mentioned, that `description_link` has higher priority than `description` and so overwrites the text in `description` 

Edit: `copyright` works as expected ...

**Comment by oparoz** [comment]
Thanks. I've just pushed a fix.

**Comment by deMattin** [comment]
Ok.
Tested also copyright and here it is a bit complicated:
`copyright_link`: If defined, marks the whole text as link to linked file.
But it is overwritten by a (external) link in markdown of the `copyright`description (as expected and should be, I think).
So if you click on text in copyright text you get the linked local file and if you click on the link text defined in markdown as link you go to "external" link.
I think it's fine this way, but you don't see, what the click on the different parts of copyright text do or that it is a difference where you click at all.
I don't know, how to describe this in simple words - but it's on the owner to design a good configuration or copyright file (color, italic, ...) and now you have any possibility ... ;)

One thing I don't like and don't know, if you can fix this:
A link in markdown text should be opened in a new browser window or tab and not in parent overwriting the ownCloud tab.

**Comment by oparoz** [comment]
I think the copyright issue is a documentation issue. I need to put up several configuration examples to illustrate how it works. The main goal of the copyright is to type in a statement which turns into a link if the file is present.
I'm trying to be flexible by allowing people to use markdown, but they have to realise that you can't use both markdown and a link together.

Links are opened in new tabs. It works for the copyright as well if used the way I described above.
If that's not the case then it's a bug and I would need an example file.

**Comment by deMattin** [comment]
This opens the google link in ownCloud tab.

``` js
---
# Gallery+ configuration file
information:
  description:
  description_link: gallery.info.md
  copyright: © 2015 MK, Germany [google](http://google.com)
  copyright_link:
  inherit: yes
sorting:
  type: name
  order: asc
  inherit: yes
```
